Disher/View Controllers/RecipesViewController.m

```diff
@@ -46,35 +56,53 @@ - (void)viewDidLoad {
 //}
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
-    return self.mealDBresults.count;
+    return self.spoonResults.count + self.mealDBresults.count;
 }
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     RecipeCell *cell = [tableView dequeueReusableCellWithIdentifier:@"RecipeCell" forIndexPath:indexPath];
-    
-    // Configure the cell...
-    cell.recipeName.text = self.mealDBresults[indexPath.row][@"strMeal"];
-    cell.recipeDescription.text = self.mealDBresults[indexPath.row][@"strArea"];
-    NSString *imageLink = self.mealDBresults[indexPath.row][@"strMealThumb"];
-    NSURL *imageURL = [NSURL URLWithString:imageLink];
-    
-    [cell.recipeImage setImageWithURL:imageURL];
+    cell.delegate = self;
+    // loop through mealdb results first, then go to spoonacular results
+    if (indexPath.row < self.mealDBresults.count) {
+        NSString *recipeName = self.mealDBresults[indexPath.row][@"strMeal"];
+        NSString *imageLink = self.mealDBresults[indexPath.row][@"strMealThumb"];
+        NSString *mealID = self.mealDBresults[indexPath.row][@"idMeal"];
+        cell.recipeName.text = recipeName;
+        NSURL *imageURL = [NSURL URLWithString:imageLink];
+        [cell.recipeImage setImageWithURL:imageURL];
+        cell.recipeSource.text = @"TheMealDB";
+        //create uniform data model for mealdb
+        Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:@"mealdb" withID:mealID];
+        [self.tableViewRecipes addObject:newRecipe];
+        cell.recipe = newRecipe;
+        cell.rightUtilityButtons = [self rightButtons];
+        
+    }
+    else {
+        NSString *recipeName = self.spoonResults[indexPath.row - self.mealDBresults.count][@"title"];
+        NSString *mealID = [NSString stringWithFormat:@"%@", self.spoonResults[indexPath.row - self.mealDBresults.count][@"id"]];
+        cell.recipeName.text = recipeName;
+        NSString *imageLink = self.spoonResults[indexPath.row - self.mealDBresults.count][@"image"];
+        NSURL *imageURL = [NSURL URLWithString:imageLink];
+        [cell.recipeImage setImageWithURL:imageURL];cell.recipeSource.text = @"Spoonacular";
+        //create uniform data model for spoonacular
+        Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:@"spoonacular" withID:mealID];
+        [self.tableViewRecipes addObject:newRecipe];
+        cell.recipe = newRecipe;
```

Comment:
There's some shared logic here that could probably be pulled outside the if/else to reduce code duplication